src/models/player.rs

```diff
@@ -1063,6 +1131,227 @@ fn push_seek_to_api<E: Env + 'static>(seek_log_req: SeekLogRequest) -> Effect {
+
+                // var durationRatio = player.length / closestDuration
+                let duration_ration = Ratio::new(library_item.state.duration, *closest_duration);
+
+                // even though we checked for len() > 0 make sure we don't panic if somebody decides to remove that check!
+                // introData = intro.gaps[closestDuration].seekHistory[0]
+                skip_gaps.seek_history.first().map(|seek_event| {
+                    IntroData {
+                        // introData.seekFrom *= durationRatio
+                        from: (duration_ration * seek_event.from).to_integer(),
+                        // introData.seekTo *= durationRatio
+                        to: (duration_ration * seek_event.to).to_integer(),
+                        // if (durationDiffInSec) {
+                        //     introData.seekDuration = introData.seekTo - introData.seekFrom
+                        // }
+                        duration: if duration_diff_in_secs > 0 { Some(seek_event.to - seek_event.from) } else { None }
+                    }
+                })
+
+              })
+                // TODO?!
+                // var correction = skipIntroCorrection()
+                // smallSkipButtonFrom = introData.seekFrom - correction.start
+                // smallSkipButtonTo = introData.seekTo + correction.end - 3000
+            };
+
+            eq_update(
+                intro_outro,
+                Some(IntroOutro {
+                    intro: intro_time,
+                    outro: outro_time,
+                }),
+            )
+        }
+        _ => Effects::none().unchanged(),
+    };
+
+    skip_gaps_effects.join(intro_outro_effects)
+}
+
+fn skip_gaps_update<E: Env + 'static>(
+    profile: &Profile,
+    selected: Option<&Selected>,
+    video_params: Option<&VideoParams>,
+    series_info: Option<&SeriesInfo>,
+    library_item: Option<&LibraryItem>,
+    skip_gaps: &mut Option<(SkipGapsRequest, Loadable<SkipGapsResponse, CtxError>)>,
+) -> Effects {
+    let active_premium = profile.auth.as_ref().and_then(|auth| {
+        auth.user
+            .premium_expire
+            .filter(|premium_expire| premium_expire > &E::now())
+            .map(|premium_expire| (premium_expire, auth.key.clone()))
+    });
+
+    let skip_gaps_request_effects = match (
+        active_premium,
+        selected,
+        video_params,
+        series_info,
+        library_item,
+    ) {
+        (
+            Some((_expires, auth_key)),
+            Some(selected),
+            Some(video_params),
+            Some(series_info),
+            Some(library_item),
+        ) => {
+            match (
+                &selected.stream.source,
+                selected.stream.name.as_ref(),
+                video_params.hash.clone(),
+            ) {
+                (StreamSource::Torrent { .. }, Some(stream_name), Some(opensubtitles_hash)) => {
```

Comment:
We should be able to handle `StreamSource::Url` (but not magnets) too

Reply:
Can you elaborate on how?
Maybe this can be subsequent PR as this PR is only focusing on adding basic support taken from the Desktop app and there we only handle Magnet links.

Reply:
I've opened #610 for this as it's not supported by desktop and needs to be tested.